core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -184,14 +184,27 @@ private[v1] class ApiRootResource extends ApiRequestContext {
   @Path("applications/{appId}/logs")
   def getEventLogs(
       @PathParam("appId") appId: String): EventLogDownloadResource = {
-    new EventLogDownloadResource(uiRoot, appId, None)
+    try {
+      // withSparkUI will throw NotFoundException if attemptId is existed for this application.
+      // So we need to try again with attempt id "1".
+      withSparkUI(appId, None) { _ =>
```

Comment:
Yes, that's true. Either we could get application acls in listing applications like what did before.

I assume download is not used very frequently, so workaround probably should be OK.